Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp

```diff
@@ -135,31 +139,101 @@ class AcceleratedBufferPool {
     RunLoop::Timer m_releaseUnusedBuffersTimer;
 };
 
+class UnacceleratedThreadedRenderingPool {
+    WTF_MAKE_FAST_ALLOCATED();
+public:
+    static UnacceleratedThreadedRenderingPool& singleton()
+    {
+        static NeverDestroyed<UnacceleratedThreadedRenderingPool> s_pool;
+        return s_pool;
+    }
+
+    UnacceleratedThreadedRenderingPool()
+        : UnacceleratedThreadedRenderingPool(numPaintingThreads())
+    {
+    }
+
+    void postPaintingTask(const IntRect& initialClip, const Ref<Nicosia::Buffer>& buffer, std::unique_ptr<DisplayList::DisplayList>&& displayList)
+    {
+        m_workerPool->postTask([initialClip, buffer, displayList = WTFMove(displayList)] {
```

Comment:
How is the buffer passed to thread here? is it captured by copying reference? I think we should make a new Ref with buffer = Ref { buffer }

Reply:
We pass on the const reference.

Reply:
Is it safe?

Reply:
Probably cleaner to seperate concerns, and create a new Ref. Will adapt.